src/enclave_proc/resource_manager.rs

```diff
@@ -905,6 +922,168 @@ impl EnclaveManager {
         Ok(())
     }
 
+    /// Set measurements field inside EnclaveHandle
+    pub fn set_measurements(
+        &mut self,
+        measurements: BTreeMap<String, String>,
+    ) -> NitroCliResult<()> {
+        self.enclave_handle
+            .lock()
+            .map_err(|e| {
+                new_nitro_cli_failure!(
+                    &format!("Failed to acquire lock: {:?}", e),
+                    NitroCliErrorEnum::LockAcquireFailure
+                )
+            })?
+            .measurements = measurements;
+        Ok(())
+    }
+
+    /// Read EIF, calculate PCR values and save them.
+    ///
+    /// Go through all the section of the EIF file, extract the headers
+    /// and compute the PCRs using the hashers, based on the section type.
+    /// Measurements will be returned as result of a separate thread executing
+    /// this function.
+    pub fn get_pcrs(eif_path: &str) -> NitroCliResult<BTreeMap<String, String>> {
```

Comment:
As a general comment for this function and `get_certificate_info` - I would not use `expect`s inside these functions, just return errors instead.